### PR TITLE
Urls: Parsing key values will not convert to numbers

### LINKED ISSUE #33890: Variables: Values with leading zeros are casted to numbers
**What happened**:

1. Having a variable with values like: `001`, `002`, etc.
2. When a value is selected from the dropdown
3. Then values are converted to numbers, e.g. `002` becomes 2

**What you expected to happen**:

Expected: handle such values as strings like in 7.5.5:

| 7.5.5 | 7.5.0-pre / main |
|------|----------------|
|  |  |

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a custom variable with values "001,002"
2. Go to the dashboard and try selecting the value, e.g. `002`
3. The dropdown will show incorrect current value, e.g. `2`. The list of values is correct, also when the value changes the URL is updated correctly, it's just the current value that is incorrect. 

**Anything else we need to know?**:

The value seems to be converted to a number with `_.toNumber()` [here](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/url.ts#L159).

**Environment**:
- Grafana version: 7.5.0-pre (a40946b6aa872d51820bc8ff14e166c5ff95d6ec)
- Data source type & version: Any
- OS Grafana is installed on: macOS 10.15.7
- User OS & Browser: Chrome
- Grafana plugins: None
- Others: N/A

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR removes the `toNumber` parsing of url params.

**Which issue(s) this PR fixes**:
Fixes #33890

**Special notes for your reviewer**:
Will create a follow up PR after v8 for cleaning up the variable system.
https://github.com/grafana/grafana/projects/14#card-60947877

### CODE REVIEW

**Comment by domasx2** [comment]
IMHO trying to guess type of query param based on value is a footgun. I had bugs were things broke in odd ways when user would enter "1" or "false" into a search field. 
Maybe we could mark current function as depreacated and introduce new one that only parses to `string` or `string[]` ?

**Comment by dprokop** [comment]
@hugohaggmark did an exploratory testing and haven't found any regressions

**Comment by hugohaggmark** [comment]
@grafana/observability-squad can you think of anything that would affect Explore? I needed to update a test for instance.

**Comment by hugohaggmark** [comment]
> @hugohaggmark did an exploratory testing and haven't found any regressions

Thank you @dprokop 🙏  then all we need is an approval 😉

**Review by dprokop — APPROVED** [review]
My bad, there you go 👍

File: packages/grafana-data/src/utils/url.ts [hunk 1]
```diff
@@ -156,7 +155,7 @@ export function parseKeyValue(keyValue: string) {
 
         let parsedVal: any;
         if (typeof val === 'string' && val !== '') {
-          parsedVal = val === 'true' || val === 'false' ? val === 'true' : toNumber(val);
+          parsedVal = val === 'true' || val === 'false' ? val === 'true' : val;
```
Comment (ifrost) [inline packages/grafana-data/src/utils/url.ts:158]: I think `number` should be removed from [UrlQueryValue](https://github.com/grafana/grafana/blob/b603a8dbb784c43a83e0316d9501cc8191add760/packages/grafana-data/src/utils/url.ts#L12) type, but then some type checks are failing, so I guess some places may require extra casting to numbers (?)

Comment (hugohaggmark) [reply packages/grafana-data/src/utils/url.ts:158]: Hmm, that is a public api. Not sure we want to change that in such a short notice.

Comment (ifrost) [reply packages/grafana-data/src/utils/url.ts:158]: Or maybe I misunderstood it. Is `UrlQueryValue` used as input (from URL) and output type (to update URL)?

**Review by ifrost — APPROVED** [review]

**Pull request merged**